bin/asc.js

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env node
+
+import childProcess from "child_process";
+
+const [ nodePath, thisPath, ...args ] = process.argv;
+const nodeArgs = process.execArgv;
+
+const hasSourceMaps = nodeArgs.includes("--enable-source-maps");
+const hasCustomArgs = args.includes("--");
+
+if (!hasSourceMaps || hasCustomArgs) {
+  if (!hasSourceMaps) {
+    nodeArgs.push("--enable-source-maps");
+  }
+  if (hasCustomArgs) {
+    const index = args.indexOf("--");
+    nodeArgs.push(...args.slice(index + 1));
+    args.length = index;
+  }
+  childProcess.spawnSync(
```

Comment:
Is it possible use async `childProcess.spawn` here?

Reply:
Unlike `fs`, there are no promise versions for `child-process` as far as I see it, so there is not much value in switching this I guess.

Reply:
I see

Reply:
Looky looky, Node has a built in function to promisify any callback API:

https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original

Also, it's easy to do manually.

```ts
// take any Node API
declare function anyNodeCallbackAPI(...args, callback) {}

await new Promise((resolve, reject) => {
  anyNodeCallbackAPI(...args, (err, result) => {
    if (err) return reject(err)
    resolve(result)
  })
})
```

That's basically all promisify does.